### PR TITLE
⭐Portfolio Project⭐ -> PopUp Window

### PR DESCRIPTION
**Pull request by Rysth**
### 🏅 Features Implemented
_Project Requirements -> https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md_

List of things that were implemented based on the requirements:

- ✅ Opened a new PopUp window when the user clicks on the action button of a Portfolio Card.
- ✅ Closed the PopUp window when the user clicks on the `x-icon` button of a Portfolio Card.
- ✅ Created an `array` with all the information about each Project using JavaScript.
- ✅ Filled each object within the `array` which contains properties with their own values.
- ✅ Updated dynamically the HTML `index.html` file with the Portfolio Cards when the page is loaded.
- ✅ Finished the **mobile** and **desktop** version of the PopUp Window.

### ⚠️ Webhint ⚠️ 

Still not working just on GitHub.

_John Palacios_
_Microverse Student_

### CODE REVIEW

File: js/scripts.js [hunk 1]
```diff
@@ -1,23 +1,242 @@
...
+
+    bodyElement.style.overflow = 'hidden';
+    bodyElement.appendChild(modalItem);
+
+    const modalCloseButton = document.querySelector('#modal__button--close');
+    modalCloseButton.addEventListener('click', () => {
+      bodyElement.removeChild(modalItem);
+      bodyElement.style.overflow = 'auto';
+    });
+  }
```
Comment (bobb-Rob) [inline js/scripts.js]: - [x] Please add a scroll-down functionality to the modal on mobile view as you did on desktop view. AS you can see in the image below, I cannot scroll done to click or tap on the buttons. 

image

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @Rysth, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)
✅ Good design

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/scripts.js (hunk 1 above)
Comment (Rysth) [reply js/scripts.js]: Thanks, @bobb-Rob!

### 🏅 Fixeds Implemented

List of things that were fixed based on the last review:

- ✅ Fixed the scroll functionality on some mobile devices.

_John Palacios_
_Microverse Student_

File: css/style.css [hunk 2]
```diff
@@ -513,6 +515,104 @@ ul {
...
+.modal__actions {
+  gap: 0.5em;
+}
+
+.modal__button {
+  flex: 1;
+  background: none;
+  font-size: 15px;
+  gap: 1em;
+}
```
Comment (Ghiftee) [inline css/style.css]: - [ ] On the desktop version, your button does not match the given design as shown below. Please ensure that the button designs match.

### Your design
Screenshot 2023-06-21 at 20 39 49 (2)

### Figma design
Screenshot 2023-06-21 at 20 39 39 (2)

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ App works as expected
✔️ App matches given design
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: css/style.css (hunk 2 above)
Comment (Rysth) [reply css/style.css]: Thanks, @Ghiftee 

### 🏅 Fixeds Implemented

List of things that were fixed based on the last review:

- ✅ Fixed the `buttons` design from the PopUp Window.

_John Palacios_
_Microverse Student_

**Review by danmainah — APPROVED** [review]
Hi @Rysth and @CobbyElsonfx , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Good design, matches the Figma design given
✔️  Ensured the popup window can be accessed using the project button , and its dynamic generated

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**